### PR TITLE
Make Rust `beta` and `nightly` experimental

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

Allow the `beta` and `nightly` checks in our build matrix to fail.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by pvdrz — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,12 +22,28 @@ on:
 
 jobs:
   build-and-test:
-    name: build and test
+    name: Build & Test ${{ format('({0},{1}){2}', matrix.os, matrix.rust, matrix.experimental && ' - 🦜' || '') }}
```
Comment (thibault-martinez) [inline .github/workflows/ci.yml]:
```suggestion
    name: Build & Test ${{ format('({0}, {1}){2}', matrix.os, matrix.rust, matrix.experimental && ' - 🦜' || '') }}
```
Do you mind if we do that ? It's stressing me out a bit.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -108,14 +122,24 @@ jobs:
           args: --workspace --doc --all-features --release
 
   no_std:
+    name: Check [no_std] ${{ format('({0},{1}){2}', matrix.rust, matrix.crates, matrix.experimental && ' - 🦜' || '') }}
```
Comment (thibault-martinez) [inline .github/workflows/ci.yml]:
```suggestion
    name: Check [no_std] ${{ format('({0}, {1}){2}', matrix.rust, matrix.crates, matrix.experimental && ' - 🦜' || '') }}
```

**Review by thibault-martinez — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -164,6 +188,7 @@ jobs:
           args: --release --target=riscv64gc-unknown-none-elf --manifest-path ${{ matrix.crates }}/Cargo.toml
 
   format:
+    name: Formatting
```
Comment (thibault-martinez) [inline .github/workflows/ci.yml]: `Formatting` looks like it's going to actually change the format...

Comment (grtlr) [reply .github/workflows/ci.yml]:
```suggestion
    name: Check Formatting
```

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Pull request merged**